test/common/protobuf/utility_test.cc

```diff
@@ -283,6 +283,23 @@ TEST_F(ProtobufUtilityTest, DEPRECATED_FEATURE_TEST(LoadBinaryV2ProtoFromFile))
   EXPECT_GT(runtime_deprecated_feature_use_.value(), 0);
 }
 
+// Verify that a config with a deprecated field can be loaded with runtime global override.
```

Comment:
A few things with this example:
1. This is bootstrap, but in reality, we need the bootstrap to check runtime, so bootstrap doesn't practically have this support. OTOH, as an example that some proto works with this feature, it's fine.
2. This test has v2, maybe it should be v3 only? I think to make that work we would have to run the script and mark some v3 fields as fatal-by-default. Is that plausible in this PR or is it more major work that makes sense in a distinct PR?

Reply:
1. Yeah, it is specific to bootstrap, and it might make more sense to add this to: `test/config_test/deprecated_configs_test.cc`. I've added it here because of the [test above it](https://github.com/envoyproxy/envoy/blob/b74f977ca789d846e0ae703a0c8c629e1dbe5dd5/test/common/protobuf/utility_test.cc#L270).
1. I agree that this should be migrated to v3 deprecated fields, along with many other tests. I typically prefer to keep PRs focused on a single item at a time (if possible). Once this is done, I'd be happy to run the script, and update this test (and all the rest). WDYT?

Reply:
SG, let's take this to the next PR.